src/lib/log-forwarding.js

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const rtLib = require('@adobe/aio-lib-runtime')
+const { writeAio, writeEnv } = require('./import')
+const crypto = require('crypto')
+const fs = require('fs-extra')
+
+const SECRET_FIELD_TYPE = 'password'
+const CHECKSUM_FILE = 'log-forwarding-config.sha256'
+
+class LogForwarding {
+  constructor (aioConfig) {
+    this.aioConfig = aioConfig
+  }
+
+  async init () {
+    rtLib.utils.checkOpenWhiskCredentials({ ow: this.aioConfig.runtime })
+    this.logForwarding = await getRTLogForwarding(this.aioConfig.runtime)
+    return this
+  }
+
+  getLocalConfig () {
+    const config = this.aioConfig.project.workspace.log_forwarding
+    try {
+      return convertToConfigObject(config)
+    } catch (e) {
+      throw new Error('Incorrect local log forwarding configuration. ' + e.message)
+    }
+  }
+
+  getLocalConfigWithSecrets () {
+    const config = this.getLocalConfig()
+    const destination = config.getDestination()
+    const settings = config.getSettings()
+    if (config.isDefined()) {
+      const destinationSettings = this.logForwarding.getDestinationSettings(destination)
+      const missingSecrets = []
+      destinationSettings.forEach(e => {
+        if (e.type === SECRET_FIELD_TYPE) {
+          const secretVarName = getSecretVarName(destination, e.name)
+          if (process.env[secretVarName] !== undefined) {
```

Comment:
this is very mysterious ... who sets this secret? how? why process.env?

Reply:
I looked at an examples like this one https://github.com/adobe/aio-lib-core-config/blob/master/src/Config.js#L51 
Is there a better way to read values from `.env` file?

Reply:
Though I agree that this code depends on the `dotenv(true)` happening before it (which happens during config load). So if there is a better way (a clear API for reading secret data from `.env` or whatever place is expected for this data), let's use it.